### PR TITLE
tearsofguthix: add timer

### PR DESCRIPTION
**Pull request by hiwilliam36**
Closes https://github.com/runelite/runelite/issues/10349
Please tell me if there is anything I need to fix, as this is my first time doing something like this.

### CODE REVIEW

**Comment by Adam-** [comment]
Is there a reason why you chose to use an overlay instead of a `Timer` infobox? They are more compact and the code which manages them largely already exists.

**Comment by hiwilliam36** [comment]
I don't really have a good reason. I used them because the tutorial I watched used an overlay so I was more comfortable using that. If you want I can try to use a timer instead.

**Comment by Adam-** [comment]
That would be better, and make it consistent with all of the other timers in the client.

**Comment by hiwilliam36** [comment]
I have changed it to use a Timer infobox. Please tell me if there is anything else I need to change.

File: runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java [hunk 1]
```diff
@@ -112,4 +142,68 @@ public void onDecorativeObjectDespawned(DecorativeObjectDespawned event)
...
+			timerRunning = false;
+		}
+	}
+
+	int getQp()
+	{
+		try
+		{
+			int qp = client.getVar(VarPlayer.QUEST_POINTS);
+			chatClient.submitQp(client.getLocalPlayer().getName(), qp);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java]: This code is for the `!qp` command. It doesn't make sense here.

**Comment by Adam-** [comment]
varbit 5099 holds the timer used to draw the in-game progress bar. It is initialized to your QP and decreases by 4 each tick. You could use that to initialize and infobox time.

File: runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java [hunk 2]
```diff
@@ -24,30 +24,38 @@
...
-import java.time.Instant;
-import java.util.HashMap;
-import java.util.Map;
-import javax.inject.Inject;
+import com.google.inject.Provides;
 import lombok.Getter;
-import net.runelite.api.Client;
-import net.runelite.api.DecorativeObject;
-import net.runelite.api.ObjectID;
+import net.runelite.api.*;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java]: Don't star-import non-static imports. [See our code conventions for more info.](https://github.com/runelite/runelite/wiki/Code-Conventions)

File: runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java [hunk 3]
```diff
@@ -112,4 +142,68 @@ public void onDecorativeObjectDespawned(DecorativeObjectDespawned event)
 		DecorativeObject object = event.getDecorativeObject();
 		streams.remove(object);
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!hasCreatedTimer)
+		{
+			if (client.getLocalPlayer().getWorldLocation().getRegionID() == 12948 && client.getLocalPlayer().getWorldLocation().getX() >= 3254 && client.getLocalPlayer().getWorldLocation().getX() <= 3262)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/tearsofguthix/TearsOfGuthixPlugin.java]: All of these worldpoint bounds should be made into a constant `WorldArea`.

**Pull request closed without merging**